executor/src/constant_evaluator/mod.rs

```diff
@@ -45,7 +48,10 @@ fn generate_values<T: FieldElement>(
     body: &FunctionValueDefinition<T>,
     index: Option<u64>,
 ) -> Vec<T> {
-    let symbols = Definitions(&analyzed.definitions);
+    let symbols = CachedSymbols {
+        symbols: &analyzed.definitions,
+        cache: Arc::new(RwLock::new(HashMap::new())),
```

Comment:
Are you using `Arc` and `RwLock` instead of `Rc` and `RefCell` to be prepared for multithreading?

Reply:
It already is multithreaded, see line 74 / 80: `.into_par_iter()`